src/Qir/Execution/QirCommandLineTool/Program.cs

```diff
@@ -0,0 +1,156 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.CommandLine;
+using System.CommandLine.Builder;
+using System.CommandLine.Help;
+using System.CommandLine.Invocation;
+using System.CommandLine.Parsing;
+using System.IO;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Quantum.Qir.Tools;
+
+namespace CommandLineCompiler
+{
+    class Program
+    {
+        private static async Task<int> Main(string[] args)
+        {
+            var buildCommand = CreateBuildCommand();
+
+            var root = new RootCommand() { buildCommand };
+            SetSubCommandAsDefault(root, buildCommand);
+            root.Description = "Command-line tool for processing QIR DLL files.";
+            root.TreatUnmatchedTokensAsErrors = true;
+
+            Console.OutputEncoding = Encoding.UTF8;
+            return await new CommandLineBuilder(root)
+                .UseDefaults()
+                .UseHelpBuilder(context => new QsHelpBuilder(context.Console))
+                .Build()
+                .InvokeAsync(args);
+        }
+
+        /// <summary>
+        /// Creates the Build command for the command line compiler, which is used to
+        /// compile the executables from a given QIR DLL.
+        /// </summary>
+        /// <returns>The Build command.</returns>
+        private static Command CreateBuildCommand()
+        {
+            var buildCommand = new Command("build", "(default) Build the executables from a QIR DLL.")
+            {
+                Handler = CommandHandler.Create((BuildOptions settings) =>
+                {
+                    if (settings.LibraryDirectories.Length < 1)
+                    {
+                        throw new ArgumentException("The '--libraryDirectories' option requires at least one argument.");
+                    }
+                    if (settings.IncludeDirectories.Length < 1)
+                    {
+                        throw new ArgumentException("The '--includeDirectories' option requires at least one argument.");
+                    }
+                    return QirTools.BuildFromQSharpDll(settings.QsharpDll, settings.LibraryDirectories[0], settings.IncludeDirectories[0], settings.ExecutablesDirectory);
```

Comment:
Why does this take the first element? If only one `--includeDirectories` option is allowed, then I would suggest enforcing that with `Option.Arity` (https://github.com/dotnet/command-line-api/blob/main/src/System.CommandLine/Option.cs#L23) or at least using `.Single()` to raise an exception if multiple options are provided.

Reply:
This is related to an earlier conversation here: https://github.com/microsoft/qsharp-runtime/pull/681#discussion_r631538735
I don't seem to have access to the constructor you linked to. I don't think we have access to Option arity at all in this code.

Reply:
This has been fixed now. The API correctly accepts an array for the libraries and headers paths. This has allow for the arguments to the command line tool to also be made optional. I also figured out how to use the Arity property, although it seems to work a little differently than the online documentation shows.